solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -2883,11 +2885,22 @@ public void publishNodeAsDown(String nodeName) {
           PerReplicaStatesOps.downReplicas(replicasToDown,
               PerReplicaStates.fetch(coll.getZNode(), zkClient, coll.getPerReplicaStates())).persist(coll.getZNode(), zkClient);
```

Comment:
Looks like `downnode` event updates the prs replica here. 
1. it means solr updates the PRS state, when node goes down?
2. Also, look like we can update state for all replicas on a node in one batch.

Reply:
1. Yes i think so, it will flip all the replicas to State.DOWN
2. Currently, it is updating it batching it per collection. That might be the best we could do, as `PerReplicaStatesOps.downReplicas` 's 2nd argument is `PerReplicaStates` which should be specific to each collection.